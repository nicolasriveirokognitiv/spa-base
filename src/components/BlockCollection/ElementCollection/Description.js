import React from 'react';
import useTestMode from 'Hooks/useTestMode';
import { Input } from 'antd';
import { toolPropTypes } from 'Constants/propTypes';
import { elementCategories } from 'Constants/global';

function Description({ label, onChange, value, defaultValue, testId }) {
    const testProps = useTestMode(testId);
    const { TextArea } = Input;

    const handleChange = e => onChange(e.target.value);

    return (
        <TextArea
            value={value}
            defaultValue={defaultValue}
            rows={4}
            placeholder={label}
            onChange={handleChange}
            data-testid="description-text"
            {...testProps}
        />
    );
}

Description.propTypes = toolPropTypes;
Description.defaultProps = {
    label       : 'Description',
    category    : elementCategories.CONTENT,
    propertyKey : 'description',
    onChange    : () => {},
};

export default Description;
