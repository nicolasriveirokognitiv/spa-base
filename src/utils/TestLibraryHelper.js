/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
