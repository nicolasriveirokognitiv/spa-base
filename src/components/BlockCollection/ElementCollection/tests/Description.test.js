import React from 'react';
import { render, fireEvent } from 'Utils/TestLibraryHelper';
import EditorContext from 'State/editorContext';
import Description from '../Description';

describe('Description', () => {
    let container;
    let props;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        props = {
            onChange : jest.fn(),
            value    : 'chunk',
        };
    });

    const renderDescription = p => render(
        <EditorContext.Provider value={{ isTestModeActive: true }}>
            <Description {...props} {...p} />
        </EditorContext.Provider>,
    );

    test('Should render Correctly', () => {
        expect(<Description />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Description />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Description should be get from props', () => {
        const { getByTestId } = renderDescription({ testId: 'description-text' });
        const elem = getByTestId('description-text');
        expect(elem.innerHTML).toBe(props.value);
    });

    test('should call onChange prop on changes in input', () => {
        const { getByTestId } = renderDescription({ testId: 'description-text' });
        fireEvent.change(getByTestId('description-text'), { target: { value: 'foo' } });
        expect(props.onChange).toHaveBeenCalledWith('foo');
    });
});
