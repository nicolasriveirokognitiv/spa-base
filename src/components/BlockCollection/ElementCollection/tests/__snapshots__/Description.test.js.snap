// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Should match snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="ant-input"
    data-testid="description-text"
    placeholder="Description"
    rows="4"
  />
</DocumentFragment>
`;
